ocaml/xapi/xapi_vm_migrate.ml

```diff
@@ -1632,6 +1632,8 @@ let migrate_send' ~__context ~vm ~dest ~live:_ ~vdi_map ~vif_map ~vgpu_map
     if (not is_intra_pool) && not copy then (
       info "Destroying VM ref=%s uuid=%s" (Ref.string_of vm) vm_uuid ;
       Xapi_vm_lifecycle.force_state_reset ~__context ~self:vm ~value:`Halted ;
+      let vtpms = Db.VM.get_VTPMs ~__context ~self:vm in
```

Comment:
You should clean up VTPMs from all `vm_and_snapshots`, which are all deleted below.